osu.Game/Overlays/UserProfileOverlay.cs

```diff
@@ -143,24 +171,27 @@ public void ShowUser(IUser user, IRulesetInfo? ruleset = null)
 
             sectionsContainer.ScrollToTop();
 
+            if (API.State.Value != APIState.Online)
```

Comment:
I don't know that this check makes sense. I can see that you've copied the `apiUser` checking logic which is used in other similar overlays, but I think there's state changes which are not covered by what you have here. ie. the API state could be "failing" but the user could still be logged in, which would cause this to never fire.

In addition, there's internal queueing of the request, which you're bypassing by adding this check. What's the intention here?

Reply:
I think based on the other code you have, using `API.IsLoggedIn` is more correct here. I may have confused you with my previous review. Generally we'd be binding to `API.State` these days, rather than `apiUser`, but given other overlays are doing this I think we can probably live with it here as you have it for now.

Reply:
Keeping it consistent between the overlays would be my preference too. The reason I have this explicit `API.State` check here is to prevent a case like this happening: 

https://github.com/ppy/osu/assets/29103029/7dbe576b-f8ca-451a-98ce-fcb31d215b31


Where we go from `Offline` -> `Connecting` (loading overlay shows) -> `Offline` (loading overlay is not hidden due to no successful request). Maybe I'm being too cautious and this really doesn't matter in the grand scheme of things, at that point the user will already have the login overlay open and can just attempt another login. If it does matter, there's probably something else that can be done instead of making this explicit check.

Reply:
Yeah I see.

Anyways, I've changed the flow to use api state instead, as that's already what `OnlineViewContainer` was using (used within this overlay) so it makes sense to match it. Seems to work well for the scenario you describe above.

Note that I've also changed the condition to fire the request in all but offline scenarios, as this matches better with the different states the API can be in.